### PR TITLE
Chore: Add NewAnonymousSignedInUser to user service

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:
This is an attempt to centralize the logic for getting an anonymous `user.SIgnedInUser`.

There are at least two known places that require an anonymous `user.SignedInUser` to be created:
* [searchV2](https://github.com/grafana/grafana/blob/717bd4a6c051d1447c9e35385a4ba176dd391c65/pkg/services/searchV2/service.go#L155-L168)
* In [#57420](https://github.com/grafana/grafana/pull/57420#discussion_r1002270870)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by nmarrs — APPROVED** [review]
LGTM

**Comment by nmarrs** [comment]
Looks like drone build is failing due to a flaky e2e test, re-running build

**Comment by ArturWierzbicki** [comment]
I also see the HTTP middleware: https://github.com/grafana/grafana/blob/6f8fcae01b3916749403dc0e2dfbdc99bde4bd4b/pkg/services/contexthandler/contexthandler.go#L204

File: pkg/services/user/userimpl/user.go [hunk 1]
```diff
@@ -254,6 +254,21 @@ func (s *Service) GetSignedInUser(ctx context.Context, query *user.GetSignedInUs
 	return signedInUser, err
 }
 
+func (s *Service) NewAnonymousSignedInUser(ctx context.Context) (*user.SignedInUser, error) {
```
Comment (ArturWierzbicki) [inline pkg/services/user/userimpl/user.go:257]: not 100% sure about this, but should we return an error if 

https://github.com/grafana/grafana/blob/94ed7444541a5cb4ce8ee13e81aa0207a3ed3631/pkg/setting/setting.go#L371

is false?

**Review by ArturWierzbicki — APPROVED** [review]

File: pkg/services/user/userimpl/user.go (hunk 1 above)
Comment (toddtreece) [reply pkg/services/user/userimpl/user.go:257]: makes sense to me. added with https://github.com/grafana/grafana/pull/57537/commits/812d146acd38362c01a0a3ca31379ba3d242061c

Comment (ArturWierzbicki) [reply pkg/services/user/userimpl/user.go:258]: Nice, lgtmm!

**Pull request merged**